airbyte-integrations/connectors/source-google-search-console/source_google_search_console/spec.json

```diff
@@ -0,0 +1,43 @@
+{
+  "documentationUrl": "https://docsurl.com",
+  "connectionSpecification": {
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "title": "Google Search Console Spec",
+    "type": "object",
+    "required": ["site_urls", "start_date"],
+    "additionalProperties": false,
+    "properties": {
+      "client_id": {
+        "type": "string",
+        "description": "Input this field if you want authorize through personal account",
+        "airbyte_secret": true
+      },
+      "client_secret": {
+        "type": "string",
+        "description": "Input this field if you want authorize through personal account",
+        "airbyte_secret": true
+      },
+      "refresh_token": {
+        "type": "string",
+        "description": "Input this field if you want authorize through personal account",
+        "airbyte_secret": true
+      },
+      "site_urls": {
+        "type": "array",
+        "description": "Website URLs property; do not include the domain-level property in the list",
+        "examples": ["https://example1.com", "https://example2.com"]
+      },
+      "start_date": {
+        "type": "string",
+        "description": "The date from which you'd like to replicate data in the format YYYY-MM-DD.",
+        "examples": ["2021-03-01"],
+        "pattern": "^[0-9]{4}-[0-9]{2}-[0-9]{2}$"
+      },
+      "service_account_info": {
```

Comment:
The right way to support multiple authentication options would be to use a `oneOf` block like described in the [connector specification reference ](https://docs.airbyte.io/connector-development/connector-specification-reference). This is more friendly to the end user as they will be shown a dropdown to choose which auth method to use, and you can also say which fields are required (whereas now we need to list all fields as optional). Let's use that method instead